mm2src/coins/lp_coins.rs

```diff
@@ -2168,18 +2183,73 @@ impl CoinsContext {
         tokens: Vec<MmCoinEnum>,
     ) -> Result<(), MmError<PlatformIsAlreadyActivatedErr>> {
         let mut coins = self.coins.lock().await;
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+
         if coins.contains_key(platform.ticker()) {
             return MmError::err(PlatformIsAlreadyActivatedErr {
                 ticker: platform.ticker().into(),
             });
         }
 
+        let platform_ticker = platform.ticker().to_string();
         coins.insert(platform.ticker().into(), platform);
 
         // Tokens can't be activated without platform coin so we can safely insert them without checking prior existence
         for token in tokens {
-            coins.insert(token.ticker().into(), token);
+            coins.insert(token.ticker().into(), token.clone());
+            platform_coin_tokens
+                .entry(platform_ticker.clone())
+                .or_default()
+                .insert(0, token.ticker().to_string());
         }
+
+        Ok(())
+    }
+
+    /// Get enabled coins to disable.
+    pub async fn get_tokens_to_disable(&self, ticker: &str) -> Vec<String> {
+        let coins = self.platform_coin_tokens.lock();
+        coins.get(ticker).cloned().unwrap_or_default()
+    }
+
+    pub async fn remove_coin(&self, ctx: &MmArc, ticker: &str) -> Result<(), String> {
+        let coin = match lp_coinfind(ctx, ticker).await {
+            Ok(Some(coin)) => coin,
+            Ok(None) => return ERR!("No such coin: {}", ticker),
+            Err(err) => return ERR!("!lp_coinfind({}): ", err),
+        };
+        let coins_ctx = try_s!(CoinsContext::from_ctx(ctx));
```

Comment:
Why move this function to `CoinsContext` implementation and then get the `coins_ctx`?

Reply:
This was my suggestion to move `remove_coin` into `CoinsContext`.
I agree with that we shouldn't get the `coins_ctx` within `CoinsContext`, but also `lp_coinfind` shouldn't be used as well.